app/src/main/cpp/BrowserWorld.cpp

```diff
@@ -766,32 +794,34 @@ BrowserWorld::AddWidget(const WidgetPlacement& placement, int32_t aCallbackId) {
   parent->GetSurfaceTextureSize(parentWidth, parentHeight);
   parent->GetWorldSize(parentWorldWith, parentWorldHeight);
 
-  float worldWidth = placement.width * kWorldDPIRatio;
-  float worldHeight;
-
-  WidgetPtr widget = Widget::Create(m.contextWeak,
-                                    placement.widgetType,
-                                    (int32_t)(placement.width * m.displayDensity),
-                                    (int32_t)(placement.height * m.displayDensity),
-                                    worldWidth);
-  widget->SetAddCallbackId(aCallbackId);
+  float worldWidth, worldHeight;
   widget->GetWorldSize(worldWidth, worldHeight);
 
-  vrb::Vector translation = vrb::Vector(placement.translation.x() * kWorldDPIRatio,
-                                        placement.translation.y() * kWorldDPIRatio,
-                                        placement.translation.z() * kWorldDPIRatio);
+  vrb::Matrix transform = vrb::Matrix::Identity();
+  if (aPlacement.rotation != 0.0) {
```

Comment:
It is possible the rotation axis is not along Y so there would still be a non-identity transform created even if rotation is 0.0f. Probably better to check if axis magnitude is near zero?